### PR TITLE
Decouple API key version from node version

### PR DESCRIPTION
**Pull request by AthenaEryma**
This commit decouples the version stored in API key documents from the node version, as part of the broader effort to stop relying on node version for product logic. The actual data format does not change - the version is stored as an integer before and after this commit, it's just determined by a manually set `ApiKey.Version` class rather than being derived from the node version.

### CODE REVIEW

**Review by jfreden — COMMENTED** [review]
Code looks good! Just a comment on the overall use of version for API keys. 

The purpose of the version field in production looks like debugging? So if there is a version change of an API key and we try to update it with an older version (could this happen in mixed clusters or after downgrades?) a debug log will happen but the update is successful. 

After this is introduced, the process of bumping the version number for API Keys when  breaking changes are introduced will be manual, which makes it easy to forget about and maybe not that useful?

If the case for the: `ApiKey.Version(currentVersionedDoc.doc().version);
        assert currentDocVersion.onOrBefore(targetDocVersion) : "current API key doc version must be on or before target version";` for CI purposes is strong enough, maybe that's enough to keep it around though, but after decoupling that one will also be less useful I think.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/ApiKey.java [hunk 1]
```diff
@@ -81,6 +81,18 @@ public String value() {
         }
     }
 
+    public record Version(int version) {
```
Comment (thecoop) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/ApiKey.java:84]: Not sure if implementing `VersionId` would be helpful here. Maybe as a marker?

**Comment by thecoop** [comment]
As its introducing a new version number here, it may be helpful to report it as a [ComponentVersionNumber](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/ComponentVersionNumber.java) for diagnostic purposes. It also needs some docs indicating how you work with the version number - how to increment it, any dependencies, etc

**Comment by thecoop** [comment]
Sure - these extra things aren't necessary for it to work here, but it helps to have some diagnostics and documented procedures in place around handling the new version numbers

**Comment by gwbrown** [comment]
@elasticmachine test this please

**Review by jfreden — APPROVED** [review]
LGTM!

**Comment by gwbrown** [comment]
@elasticmachine update branch

**Pull request merged**